### PR TITLE
feat(@toss/utils): Add `isNil` util function

### PR DESCRIPTION
**Pull request by SEOKKAMONI**
## Overview
Add `isNil` util function

## PR Checklist

- [x] I read and included theses actions below

1. I have read the [Contributing Guide](https://github.com/toss/slash/blob/main/.github/CONTRIBUTING.md)
2. I have written documents and tests, if needed.

### CODE REVIEW

File: packages/common/utils/src/isNil.ko.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# isNil
+
+주어진 값이 null이나 undefined인지 확인합니다.
```
Comment (SEOKKAMONI) [inline packages/common/utils/src/isNil.ko.md:3]:
```suggestion
주어진 값이 null이나 undefined 인지 확인합니다.
```

File: packages/common/utils/src/isNil.spec.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import { isNil } from '.';
+
+describe('isNil should', () => {
```
Comment (SEOKKAMONI) [inline packages/common/utils/src/isNil.spec.ts:3]:
```suggestion
describe('isNil', () => {
```

**Comment by ssi02014** [comment]
@SEOKKAMONI  👋
 I'm not a maintainer of slash, but here's a quick comment. 

Type narrowing is not working as intended.

```tsx
const test = (arg: any) => {
  if (isNil(arg)) {
    arg; // any, Should be type inferred as `null | undefined`.
  } else {
    arg; // never
  }
};
```

With improvements like the ones below, you'll be able to do type narrowing as intended. 

```ts
function isNil<T>(val: T | undefined | null): val is null | undefined {
  return val == null;
}
```
```ts
const test = (arg: any) => {
  if (isNil(arg)) {
    arg; // null | undefined
  } else {
    arg; // any
  }
};
```

File: packages/common/utils/src/isNil.ts [hunk 3]
```diff
@@ -0,0 +1,4 @@
+/** @tossdocs-ignore */
+export function isNil<T>(val: T | undefined | null): val is T {
```
Comment (SEOKKAMONI) [inline packages/common/utils/src/isNil.ts:2]:
```suggestion
export function isNil<T>(val: T | undefined | null): val is null | undefined {
```

File: packages/common/utils/src/isNil.en.md [hunk 4]
```diff
@@ -0,0 +1,15 @@
+# isNil
+
+Checks if the given value is `null` or `undefined`;
+
+```typescript
+function isNil<T>(value: T | null | undefined): value is T;
```
Comment (ssi02014) [inline packages/common/utils/src/isNil.en.md:6]:
```suggestion
function isNil<T>(value: T | null | undefined): value is null | undefined;
```

File: packages/common/utils/src/isNil.ko.md [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+
+```typescript
+function isNil<T>(value: T | null | undefined): value is T;
```
Comment (ssi02014) [inline packages/common/utils/src/isNil.ko.md:6]:
```suggestion
function isNil<T>(value: T | null | undefined): value is null | undefined;
```

**Review by ssi02014 — COMMENTED** [review]
looks good! 🤗
However, a final review by the maintainer of `slash` is required. 

The documentation file(.md) also looks like it needs to be fixed

**Review by okinawaa — APPROVED** [review]
thanks😊

Whether `isNil` will really increase the developer experience over `==` is a matter of consideration.

I think it's right to give the service developer the option to use.

**Pull request merged**